db/migrate/20150624004126_add_invocation.rb

```diff
@@ -0,0 +1,49 @@
+class AddInvocation< ActiveRecord::Migration
+  def change
+    create_table :targetings do |t|
+      t.string :search_query
+      t.references :bookmark
+      t.references :user
+      t.timestamps
+    end
+
+    add_index :targetings, [:bookmark_id, :user_id], :name => "targetings_bookmark_user_ids"
+    add_foreign_key :targetings, :bookmarks, :name => "targetings_bookmark_id", :column => 'bookmark_id'
+    add_foreign_key :targetings, :user, :name => "targetings_user_id", :column => 'user_id'
+
+    create_table :targeting_hosts do |t|
```

Comment:
nitpick: I find it bad experience to create many operations in one migration, its usually harder to fix issues if it worked half way ( I know its wrapped in a transaction) but having smaller migrations makes imho sense.